### PR TITLE
added ClearScroll action

### PR DESCRIPTION
**Pull request by pulse00**
see https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html#_clear_scroll_api

I'm working on jest support for spring-data-elasticsearch. The jest API misses this elasticsearch action, see https://github.com/spring-projects/spring-data-elasticsearch/blob/master/src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java#L562

### CODE REVIEW

**Comment by kramer** [comment]
Hi @pulse00,
A few things that, I believe, would improve this new (to jest) `ClearScroll` action:

* Builder takes only a single scroll ID whereas the original API can take multiple IDs.
* I would favor ditching the `MAX_SCROLL_ID_LENGTH` logic, which complicates things, and always using request body to send scroll id data.
* The API docs does not specify but does ES support specifying indices for this request type? If not then using `AbstractMultiIndexActionBuilder` as the builder base would be misleading.

What do you think?

**Comment by pulse00** [comment]
Hey @kramer,

> Builder takes only a single scroll ID whereas the original API can take multiple IDs.

makes sense. Do you have any existing api calls that serialize the body to an array? From what i can tell in the elasticsearch source code they use [Base64.encode/decode](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/action/search/TransportSearchHelper.java#L87) API. 




> I would favor ditching the MAX_SCROLL_ID_LENGTH logic, which complicates things, and always using request body to send scroll id data

will do.


> The API docs does not specify but does ES support specifying indices for this request type? If not then using AbstractMultiIndexActionBuilder as the builder base would be misleading.

What's the recommended base class for such a call?

**Comment by kramer** [comment]
> Do you have any existing api calls that serialize the body to an array? From what i can tell in the elasticsearch source code they use Base64.encode/decode API.

It's just the HTTP body, you don't need to encode anything, simply set the `payload` variable and [AbstractActon](https://github.com/searchbox-io/Jest/blob/master/jest-common/src/main/java/io/searchbox/action/AbstractAction.java#L149) will take care of it. For example:

```java
    protected ClearScroll(Builder builder) {
        super(builder);
        this.payload = ImmutableMap.of("scroll_id", builder.getScrollIds()); // where builder.getScrollIds() returns a list of strings
        setURI(buildURI());
    }
```

> What's the recommended base class for such a call?

I'd say: `public static class Builder extends GenericResultAbstractAction.Builder<ClearScroll, Builder>`

**Comment by kramer** [comment]
Oh and for the multiple scroll ids in the builder please try to keep the *collection behaviour* [AbstractMultiIndexActionBuilder](https://github.com/searchbox-io/Jest/blob/master/jest-common/src/main/java/io/searchbox/action/AbstractMultiIndexActionBuilder.java#L14) has, i.e.: internal set of strings, has no param constructor, has methods to add single or multiple scroll ids. Another part of that behaviour is when no ids are given it should call with `_all`appended to uri and without any payload, to do that you'll need to modify the constructor example I gave earlier (with the `ImmutableMap.of...` line).

**Comment by pulse00** [comment]
@kramer i've implemented the changes. thanks for your feedback.

File: jest-common/src/main/java/io/searchbox/core/ClearScroll.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  public static class Builder extends GenericResultAbstractAction.Builder<ClearScroll, Builder> {
+
+    protected Set<String> scrollIds = new LinkedHashSet<String>();
+
+    public Builder addScrollId(String scrollId) {
+      this.scrollIds.add(scrollId);
+      return this;
+    }
+
+    public Builder addScrollIds(Set<String> scrollIds) {
```
Comment (kramer) [inline jest-common/src/main/java/io/searchbox/core/ClearScroll.java]: Can be `Collection<? extends String>` for the sake of being more accepting.

Comment (pulse00) [inline jest-common/src/main/java/io/searchbox/core/ClearScroll.java]: done

**Comment by kramer** [comment]
Great, thanks for your contribution @pulse00 !

**Comment by pulse00** [comment]
you're welcome!

**Pull request merged**